File: DnDManagementSystem/Back-End/DnD_SpringBoot_v2/src/main/java/com/revature/models/charactersheet/Spell.java
package com.revature.models.charactersheet;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity

//Lombok
@Getter
@Setter
@NoArgsConstructor
//@AllArgsConstructor
@ToString
@EqualsAndHashCode
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "spellId")
@JsonIgnoreProperties(value = {"hibernateLazyInitializer", "handler"})

@Table(name = "spells_db")

public class Spell {
	@Id
	@Column(name = "spellId") 
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	int spellId;
	
	@Column(name = "name")
	String name;
	@Column(name = "level")
	int level;
	@Column(name = "castingTime")
	String castingTime;
	@Column(name = "rangee")
	String range;
	@Column(name = "duration")
	String duration;
	@Column(name = "school")
	String school;
	@Column(name = "save")
	String save;
	@Column(name = "damage")
	String damage;
	@Column(name = "description")
	String description;


	@ManyToOne
	@JoinColumn(name = "spellsSheetId")
	private SpellSheet spellSheet;

}

File: DnDManagementSystem/Front-End/DnDManager/src/app/models/temp-models/Spell.ts
export class Spell {
  constructor(public _name:string, public _level:number, public _castingTime:string, public _range:string,
              public _components:string, public _duration:string, public _school:string, public _save:string,
              public _damage:string, public _description:string) { }


  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get level(): number {
    return this._level;
  }

  set level(value: number) {
    this._level = value;
  }

  get castingTime(): string {
    return this._castingTime;
  }

  set castingTime(value: string) {
    this._castingTime = value;
  }

  get range(): string {
    return this._range;
  }

  set range(value: string) {
    this._range = value;
  }

  get components(): string {
    return this._components;
  }

  set components(value: string) {
    this._components = value;
  }

  get duration(): string {
    return this._duration;
  }

  set duration(value: string) {
    this._duration = value;
  }

  get school(): string {
    return this._school;
  }

  set school(value: string) {
    this._school = value;
  }

  get save(): string {
    return this._save;
  }

  set save(value: string) {
    this._save = value;
  }

  get damage(): string {
    return this._damage;
  }

  set damage(value: string) {
    this._damage = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }
}
